acasclient/ddict.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import unicode_literals
+
+import logging
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+class DDict(object):
+    """The DDict class is meant as a generic interface for any implementation of a Data Dictionary that
+    can be referenced by an ACAS CodeValue.
+    Any classes implementing this interface must implement the get_values() method.
```

Comment:
What do you think about documenting that the caller should ensure `get_values` is called before calling `check_value`?

If the caller forgets to call `get_values` and directly calls `check_value`, they might get a `TypeError` because of `valid_values` being `None`.

Reply:
Great suggestion, adding this both to the docs and adding an explicit exception if someone tries to call `check_value` without `valid_values` populated.